### PR TITLE
[DMS-309] Appropriate message to report The document ID provided is invalid.

### PR DESCRIPTION
**Pull request by JBrenesSimpat**
Example of the response obtained in ODS 7.2
image

### CODE REVIEW

File: src/core/EdFi.DataManagementService.Core/Middleware/ParsePathMiddleware.cs [hunk 1]
```diff
@@ -2,21 +2,23 @@
...
 
 internal record PathInfo(string ProjectNamespace, string EndpointName, string? DocumentUuid);
 
 /// <summary>
-/// Parses and validates the path from the frontend is well formed. Adds PathComponents
+/// Parses and validates the path from the frontend is well-formed. Adds PathComponents
 /// to the context if it is.
 /// </summary>
-internal partial class ParsePathMiddleware(ILogger _logger) : IPipelineStep
+internal class ParsePathMiddleware(ILogger logger) : IPipelineStep
```
Comment (simpat-adam) [inline src/core/EdFi.DataManagementService.Core/Middleware/ParsePathMiddleware.cs:21]: I think we agreed as a team early on to use `_logger` here even though it gives style warnings. Unless we change it everywhere I would revert this.

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/Descriptors/UpdateDescriptorsValidation.feature [hunk 2]
```diff
@@ -568,8 +568,6 @@ Feature: Update a Descriptor
                   }
                   """
 
-        # DMS-309
-        @ignore
         Scenario: 20 Verify response code 404 when ID is not valid
```
Comment (simpat-adam) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/Descriptors/UpdateDescriptorsValidation.feature:571]: Need to rename this

File: src/core/EdFi.DataManagementService.Core/Middleware/ParsePathMiddleware.cs [hunk 3]
```diff
@@ -50,33 +52,58 @@ private static bool IsDocumentUuidWellFormed(string documentUuidString)
...
-            _logger.LogDebug(
+            logger.LogDebug(
                 "ParsePathMiddleware: Not a valid path - {TraceId}",
                 context.FrontendRequest.TraceId
             );
-            context.FrontendResponse = new FrontendResponse(StatusCode: 404, Body: "", Headers: []);
+            context.FrontendResponse = new FrontendResponse(
+                StatusCode: 404,
+                Body: FailureResponse.ForInvalidPath(
+                    detail: "The path provided is invalid.",
```
Comment (simpat-adam) [inline src/core/EdFi.DataManagementService.Core/Middleware/ParsePathMiddleware.cs:68]: I don't see this in any tests. Was this change necessary as part of this ticket?

**Review by simpat-adam — CHANGES_REQUESTED** [review]

File: src/core/EdFi.DataManagementService.Core/Middleware/ParsePathMiddleware.cs (hunk 3 above)
Comment (JBrenesSimpat) [reply src/core/EdFi.DataManagementService.Core/Middleware/ParsePathMiddleware.cs:68]: You are right; the ticket does not say anything about it, so I returned the code as it was.

**Review by simpat-adam — APPROVED** [review]
